sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -75,31 +112,46 @@ export interface MessageHandlerOptions {
     maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
-
 // @public
-export interface OnError {
-    (error: MessagingError | Error): void;
+export interface MessageHandlers<ContextT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
 }
 
 // @public
-export interface OnMessage {
-    (message: ServiceBusMessage): Promise<void>;
-}
+export type MessageIterator<ContextT> = AsyncIterable<MessageAndContext<ContextT>>;
+
+export { MessagingError }
 
-// Warning: (ae-forgotten-export) The symbol "Client" needs to be exported by the entry point index.d.ts
-//
 // @public
-export class QueueClient implements Client {
+export interface NonSessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
     close(): Promise<void>;
-    createReceiver(receiveMode: ReceiveMode): Receiver;
-    createReceiver(receiveMode: ReceiveMode, sessionOptions: SessionReceiverOptions): SessionReceiver;
-    createSender(): Sender;
-    readonly entityPath: string;
-    static getDeadLetterQueuePath(queueName: string): string;
-    readonly id: string;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
+    diagnostics: {
+        peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+        peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    };
+    // (undocumented)
+    entityPath: string;
+    // (undocumented)
+    entityType: "queue" | "subscription";
+    // (undocumented)
+    getDeadLetterPath(): string;
```

Comment:
It is odd that we have this on the receiver meaning if one wants to receive from a dead letter queue, we are asking them first to create a receiver to the queue, use this helper and then create the receiver to the dead letter queue.

I'd expect this to be a static helper method on the client.
Also, it turns out in .Net Track 1 there are 4 other helpers like this.
So, its worth checking them out as well. Look for `EntityNameFormatter`

Reply:
I do hate this but the entity name formatter seems a bit worse to me.

One option I floated (but we haven't really discussed too thoroughly) was making deadLetter a flag in options or something similar.

Reply:
I like the idea of having an options flag or even a separate method like GetDeadLetterReceiver.

Reply:
I'd talked with @ramya-rao-a  about a similar idea of just having a top level "gerDeadLetterReceiver" method. I'm for it and we should talk about it for the next preview. :)

Reply:
Hated it so much it's been moved to `ServiceBusClient.getDeadLetterReceiver`

Reply:
Yeah I like getDeadLetterReceiver a lot. Count .NET in for this change as well for the next preview.